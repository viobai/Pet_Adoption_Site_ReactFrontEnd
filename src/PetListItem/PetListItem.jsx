import React from 'react';
import { Link } from "react-router-dom";
import PropTypes from "prop-types";
import style from "./PetListItem.module.css";

const PetListItem = (props) => {
    const petLink = "/id/" + props.pet.id;
    const imgDesc = "a " + props.pet.category + " of breed " + props.pet.breed + " named " + props.pet.name;
    return (
        <Link className={style.link} to={petLink}>
            <div className={style.card}>
                <p className={style.hiddenMsg}>LEARN MORE</p>
                <div className={style.imgContainer}><img src={props.pet.imageUrl} alt={imgDesc}/></div>
                <div className={style.shortInfo}>
                    <span className={style.name}>{props.pet.name}</span> {props.pet.age}
                </div> 
            </div>
        </Link>
    )
}

PetListItem.propTypes = {
    pet: PropTypes.object.isRequired
}

export default PetListItem