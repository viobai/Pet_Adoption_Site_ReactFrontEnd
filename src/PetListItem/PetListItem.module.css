.card {
  border: 16px white solid;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
  height: 50vh;
  padding: 0;
  position: relative;
}

.imgContainer {
  height: 80%;
  width: 100%;
  overflow: auto;
  background-color: black;
}

img {
  width: 100%;
  height: 100%;
  object-fit: cover;
  display: block;
  margin-left: auto;
  margin-right: auto;
  overflow:auto;
  background-repeat: no-repeat;
}

.shortInfo {
  display: flex;
  align-items: baseline;
  flex-wrap: wrap;
  align-content: center;
  padding: 0 2rem;
  /* border: orange solid; */
  height: 18.5%;
}

.link {
  color: black;
  text-decoration: none;
}

.name {
  font-size: 30px;
  font-weight: bold;
  margin-right: 1rem;
}

.hiddenMsg {
  display: none;
  position: absolute;
  top: 50%;
  left: 40%;
  transform: translate(-50%, -50%);
  color: white;
  font-size: 30px;
  font-weight: bolder;
}

.card:hover {
  border-color: rgba(0, 0, 0, .1);
  background-color: rgba(0, 0, 0, .2);
  opacity: 1;
  z-index: 2;
}

.card:hover img {
  opacity: 0.5;
}

.card:hover .hiddenMsg {
  display: block;
  z-index: 2;
}