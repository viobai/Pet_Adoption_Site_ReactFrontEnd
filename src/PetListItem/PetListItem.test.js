import React from "react";
import PetListItem from "./PetListItem";

describe("PetListItem tests", () => {
    var petDetail = {
        "id": "11111",
        "name": "Apollo",
        "category": "dog",
        "gender": "male",
        "age": "1 year 3 months",
        "size": "61.2 pounds",
        "breed": "Mixed Breed",
        "description": "<p>I can't wait to settle in to my new home. I've heard all about this family dog business and I think it sounds great - walks, playing with toys, treats, belly rubs... I'm excited about all of that! If you're as ready as I am, let's meet!</p>",
        "neutering": "Neutered",
        "characteristics": ["I can be a little bit of an escape artist! I need a yard with a 6 ft wood fence.", "I need to meet any other dogs I might be living with."],
        "imageUrl": "https://raw.githubusercontent.com/viobai/seainfo6150-final-project-webapp/master/src/images/Pet/Apollo.jpeg"
    };

    it("renders correctly", () => {  
        const { container } = render(<PetListItem pet={petDetail}/>);
        expect(container).toMatchSnapshot();
    });
});