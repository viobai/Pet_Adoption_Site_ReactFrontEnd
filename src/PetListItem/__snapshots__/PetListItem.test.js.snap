// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PetListItem tests renders correctly 1`] = `
<div>
  <a
    class="link"
    href="/id/11111"
  >
    <div
      class="card"
    >
      <p
        class="hiddenMsg"
      >
        LEARN MORE
      </p>
      <div
        class="imgContainer"
      >
        <img
          alt="a dog of breed Mixed Breed named Apollo"
          src="https://raw.githubusercontent.com/viobai/seainfo6150-final-project-webapp/master/src/images/Pet/Apollo.jpeg"
        />
      </div>
      <div
        class="shortInfo"
      >
        <span
          class="name"
        >
          Apollo
        </span>
         
        1 year 3 months
      </div>
    </div>
  </a>
</div>
`;
